src/commands/git.rs

```diff
@@ -452,10 +455,17 @@ fn cmd_git_clone(
 fn do_git_clone(
     ui: &mut Ui,
     command: &CommandHelper,
+    colocate: bool,
     source: &str,
     wc_path: &Path,
 ) -> Result<(WorkspaceCommandHelper, Option<String>), CommandError> {
-    let (workspace, repo) = Workspace::init_internal_git(command.settings(), wc_path)?;
+    let (workspace, repo) = if colocate {
+        let git_path = &wc_path.join(".git");
+        git2::Repository::init(git_path)?;
```

Comment:
I would put `git2::Repository::init(wc_path)?;` here, since initializing a non-bare repo should create the `.git` directory. OTOH, it seems to pass tests as is, so maybe I'm confused.

I'd still try it, unless you know it to be wrong. Who knows, maybe it's related to the Windows issue.

Reply:
Thanks, tried that approach, and it seems to work.

I've also removed the manual joining of `.git` to instead rely on `git2::Repository::path` method

Reply:
Thanks, that did the trick!

Reply:
Glad to hear it!

I'm surprised that the original version worked. Perhaps that's a bug in libgit2, or perhaps it's some complicated intended behavior.